### DISCUSSION TITLE
Multiple instructors creating assignments?

### DISCUSSION THREAD

**Discussion by jdmansour** [jupyter/nbgrader#1898] [General]
Hi, I wonder if there is a way to have multiple instructors work on one course, running on JupyterHub. I found the instructions for [multiple graders](https://nbgrader.readthedocs.io/en/stable/configuration/jupyterhub_config.html#example-use-case-one-class-multiple-graders), but I think it applies only to grading, not to course creation and releasing.

I tried to set it up naively, but the problem is that each instructor has their own `gradebook.db`. If I release an assignment as one instructor, the other instructor cannot perform autograding (they can however pull submissions from the exchange). I tried to copy the `release` files to the second instructor, and then add the assignments to the DB, but that also doesn't work. It seems the only way that an assignment can get added to the DB properly is by running `nbgrader generate_assignment`.

What I'd ideally like to do is to have one instructor run `generate_assignment`, `release`, and then another instructor obtain the identical sources (via git or some other way) and be able to run `autograde` etc.. Actually, one of the instructors is not a human but an automatic service that autogrades as soon as the student submits, and then reports a score to a database. The issue came up when trying to implement this, but it also applies to the simple case of two human instructors. Is anybody doing something like this or has any ideas?

**Comment by jeflem** [comment]
I'm using nbgrader on a JupyterHub together with a colleague. We have a separate user account which runs the formgrader service in JHub (one account/service per course). Both instructors have access to this account and can do everything there. Whether this is a solution for you heavily depends on your overall setup.

Usual workflow for us is, the my colleague creates assignments. Then I have a look at them, implement minor corrections and release them. If deadline is over my colleague collects the submission, does the (auto)grading.

Of course, creating a new user account for each course might be a lot of work, but for us course creation and student enrolement is done via LTI/LMS (cf. [Ananke Jupyter Distribution](https://gauss.whz.de/ananke)).

**Comment by lahwaacz** [comment]
> I found the instructions for multiple graders, but I think it applies only to grading, not to course creation and releasing.

It applies to all cases where you need a common _gradebook_ for one course.
